ci-operator/step-registry/telcov10n/sno-hub-ztp/set-hub-preserved/telcov10n-sno-hub-ztp-set-hub-preserved-commands.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "************ telcov10n cluster setup via agent command ************"
+# Fix user IDs in a container
+[ -e "${HOME}/fix_uid.sh" ] && "${HOME}/fix_uid.sh" || echo "${HOME}/fix_uid.sh was not found" >&2
+
+function setup_aux_host_ssh_access {
+
+  echo "************ telcov10n Setup AUX_HOST SSH access ************"
+
+  SSHOPTS=(
+    -o 'ConnectTimeout=5'
+    -o 'StrictHostKeyChecking=no'
+    -o 'UserKnownHostsFile=/dev/null'
+    -o 'ServerAliveInterval=90'
+    -o LogLevel=ERROR
+    -i "${CLUSTER_PROFILE_DIR}/ssh-key"
+  )
+
+}
+
+function set_expiration_time {
+
+  echo "************ telcov10n Set Hub cluster expiration time ************"
+
+  echo
+  set -x
+  b64_exp_time=$(echo "${EXPIRATION_TIME}" | base64 -w 0)
+  timeout -s 9 10m ssh "${SSHOPTS[@]}" "root@${AUX_HOST}" bash -s --  \
+  "${NAMESPACE}" "${b64_exp_time}" "${EXPIRATION_TIME_FILE}" "${SHARED_HUB_CLUSTER_PROFILE}" << 'EOF'
+set -o nounset
+set -o errexit
+set -o pipefail
+
+exp_time_ns_file=/var/builds/${1}
+exp_time=$(echo ${2} | base64 -d)
```

Comment:
do we need base64 encoding for this string?

Reply:
The problem is that the pass in parameter is like '5 days' (note the blank space in between) and whenever I tried to pass in it as text string the quotes are removed and they are pass in as 2 parameters instead of just one to be used at this line:

`TZ=UTC date --iso-8601=seconds **-d "${exp_time}"** > ${exp_time_ns_file}`

Reply:
It should work by chaining single-quotes

```
  timeout -s 9 10m ssh "${SSHOPTS[@]}" "root@${AUX_HOST}" bash -s --  \
  "${NAMESPACE}" "'${b64_exp_time}'"


Also,  "${EXPIRATION_TIME_FILE}" is not needed if you write into /var/builds/${NAMESPACE}. The less the env vars in the step, the better.

Reply:
Finally, I would generate the iso 8601 expiration date in the pod instead of the remote node.